source/material_model/visco_plastic.cc

```diff
@@ -240,9 +240,18 @@ namespace aspect
 
           // Step 3b: calculate weakened friction, cohesion, and pre-yield viscosity
           const double current_cohesion = drucker_prager_parameters.cohesions[j] * weakening_factors[0];
-          const double current_friction = drucker_prager_parameters.angles_internal_friction[j] * weakening_factors[1];
+          double current_friction = drucker_prager_parameters.angles_internal_friction[j] * weakening_factors[1];
           viscosity_pre_yield *= weakening_factors[2];
 
+          // Step 3d: calculate steady-state coefficient of internal friction
+          if (use_steady_state_friction_angle == true)
```

Comment:
What happens if you remove this if-loop completely? As far as I can see the return value of `compute_steady_state_friction` for `drucker_prager_parameters.angles_internal_friction_dynamic[j] == 0` should be exactly `current_friction` and all tests should continue to work. If there are problems due to floating-point inaccuracies, we could introduce a bypass into `compute_steady_state_friction` (e.g. `if (angle_internal_friction_dynamic == 0.0) return angle_internal_friction;`.
This would allow you to remove the additional variable for visco-plastic (which is in essence just a short-hand for `angle_internal_friction_dynamic != 0.0` anyway).